src/System.Json/src/System/Json/JsonObject.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using JsonPair = System.Collections.Generic.KeyValuePair<string, System.Json.JsonValue>;
+using JsonPairEnumerable = System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Json.JsonValue>>;
+
+namespace System.Json
+{
+    public class JsonObject : JsonValue, IDictionary<string, JsonValue>, ICollection<JsonPair>
```

Comment:
We could consider it, but my understanding is we're only really adding this assembly for compatibility with Mono (it doesn't exist on desktop), and that we'd suggest developers write new JSON-related code using Json.NET and/or https://github.com/dotnet/corefxlab/tree/master/src/System.Text.Json depending on how that experiment turns out, rather than writing new code using this library.  So I'm not sure how much effort we want to expend on adding APIs, interface implementations, etc.  We could certainly look into it, though, but I'd like that to be a separate effort it if happens.